### DISCUSSION TITLE
What is the correct way to couple a js class to a livewire component?

### DISCUSSION THREAD

**Discussion by digitall-it** [livewire/livewire#8576] [1. Help]
TL;DR: how to bind js class to livewire w/o using Alpine, Async Alpine with the ESM module called on the fly, and/or js in the body, also account for lazy loading and morphs?

Problem:
I have a livewire component called newsTicker, and I have a standard vite setup that creates a javascript bundle. In the app.js I am importing the newsTicker javascript class from news-ticker.js. I now need to create an instance of the class and connect it to the component, so that I can have an instance of the javascript class for each livewire instance. I would like not to use Alpine.js but use pure javascript, I would not want to but the javascript in the body of the HTML. My current solution involves sending custom events. I've read about ax-load and Async Alpine but would that means I will have to separately load my module? What is the "graced" way to do this? In the documentation it is not very clear. I also need to respond to lazy loads and DOM morphs/refreshes of the component.
